library/lib/core/application.cpp

```diff
@@ -317,13 +358,32 @@ void Application::onControllerButtonPressed(enum ControllerButton button, bool r
     }
 }
 
+bool Application::setInputType(InputType type)
+{
+    if (type == Application::inputType)
+        return false;
+
+    Application::inputType = type;
+
+    if (type == InputType::GAMEPAD)
+    {
+        Application::currentFocus->onFocusGained();
+    }
+
+    return true;
+}
+
 View* Application::getCurrentFocus()
 {
     return Application::currentFocus;
 }
 
 bool Application::handleAction(char button)
 {
+    // Dismiss if input type was changed
```

Comment:
Same question here, is it actually like this in HOS?

Reply:
Yes, it is.